tesseract_process_managers/src/task_generators/upsample_trajectory_task_generator.cpp

```diff
@@ -0,0 +1,155 @@
+  {
+    CONSOLE_BRIDGE_logError("Input seed to SeedMinLengthTaskGenerator must be a composite instruction");
+    saveOutputs(*info, input);
+    info->elapsed_time = timer.elapsedSeconds();
+    return 0;
+  }
+
+  // Get Composite Profile
+  const auto& ci = input_results->as<CompositeInstruction>();
+  std::string profile = ci.getProfile();
+  profile = getProfileString(name_, profile, input.composite_profile_remapping);
+  auto cur_composite_profile = getProfile<UpsampleTrajectoryProfile>(
+      name_, profile, *input.profiles, std::make_shared<UpsampleTrajectoryProfile>());
+  cur_composite_profile = applyProfileOverrides(name_, profile, cur_composite_profile, ci.profile_overrides);
+
+  assert(cur_composite_profile->longest_valid_segment_length > 0);
+  auto& results = input_results->as<CompositeInstruction>();
+  Instruction start_instruction = results.getStartInstruction();
+  CompositeInstruction new_results(results);
+  new_results.clear();
+
+  upsample(new_results, results, start_instruction, cur_composite_profile->longest_valid_segment_length);
+  results = new_results;
+
+  info->return_value = 1;
+  saveOutputs(*info, input);
+  info->elapsed_time = timer.elapsedSeconds();
+  return 1;
+}
+
+void UpsampleTrajectoryTaskGenerator::process(TaskInput input, std::size_t unique_id) const
+{
+  conditionalProcess(input, unique_id);
+}
+
+void UpsampleTrajectoryTaskGenerator::upsample(CompositeInstruction& composite,
+                                               const CompositeInstruction& current_composite,
+                                               Instruction& start_instruction,
+                                               double longest_valid_segment_length) const
+{
+  for (const Instruction& i : current_composite)
+  {
+    assert(!isPlanInstruction(i));
+    if (isCompositeInstruction(i))
+    {
+      const auto& cc = i.as<CompositeInstruction>();
+      CompositeInstruction new_cc(cc);
+      new_cc.clear();
+
+      upsample(new_cc, cc, start_instruction, longest_valid_segment_length);
+      composite.push_back(new_cc);
+    }
+    else if (isMoveInstruction(i))
+    {
+      assert(isMoveInstruction(start_instruction));
+      const auto& mi0 = start_instruction.as<MoveInstruction>();
+      const auto& mi1 = i.as<MoveInstruction>();
+
+      assert(isStateWaypoint(mi0.getWaypoint()));
+      assert(isStateWaypoint(mi1.getWaypoint()));
+      const auto& swp0 = mi0.getWaypoint().as<StateWaypoint>();
+      const auto& swp1 = mi1.getWaypoint().as<StateWaypoint>();
+
+      double dist = (swp1.position - swp0.position).norm();
+      if (dist > longest_valid_segment_length)
+      {
+        long cnt = static_cast<long>(std::ceil(dist / longest_valid_segment_length)) + 1;
+
+        // Linearly interpolate in joint space
+        Eigen::MatrixXd states = interpolate(swp0.position, swp1.position, cnt);
+
+        // Convert to MoveInstructions
+        for (long i = 1; i < states.cols(); ++i)
```

Comment:
Perhaps a comment here describing why i = 1, and why it's not states.cols() - 1

Reply:
Done.